scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -977,13 +978,18 @@ void pingTaskNode(RunningTaskData taskData) {
         } catch (Throwable t) {
             RunningTaskData runningTask = jobs.getRunningTask(task.getId());
             if (runningTask != null) {
-                int attempts = runningTask.increaseAndGetPingAttempts();
-                if (attempts > PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS.getValueAsInt()) {
-                    tlogger.info(task.getId(), "node failed", t);
-                    restartTaskOnNodeFailure(task);
+                if (NodesPinger.pingNode(runningTask.getNodes().get(0)) == -1) {
+                    int attempts = runningTask.increaseAndGetPingAttempts();
+                    if (attempts > PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS.getValueAsInt()) {
```

Comment:
With my change, the property PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS is taken into consideration only if the node is failed. Which seems to me in adequation with the property name. If the node is failed, then the number of ping attempts will increase until it reaches the maximum number of ping attemps, meaning it will not wait forever. And then the task will initiate a restart and increase the failure counter.